README.md

```diff
@@ -433,6 +433,203 @@ to confirm that our debugging object is still displayed:
+/* inside the App component `make` function */
+    let (data, setData) = React.useState(() => Loading);
+    React.useEffect0(() => {
+      Js.Promise.(
+        Fetch.fetch("https://gh-feed.vercel.app/api?user=jchavarri&page=1")
+        |> then_(Fetch.Response.text)
+        |> then_(text =>
+             {
+               let data =
+                 try(Ok(text |> Json.parseOrRaise |> Feed.feed_of_json)) {
+                 | Json.Decode.DecodeError(msg) =>
+                   Js.Console.error(msg);
+                   Error("Failed to decode: " ++ msg);
+                 };
+               setData(_ => Loaded(data));
+             }
+             |> resolve
+           )
+      )
+      |> ignore;
+      None;
+    });
+```
+
+### 7.2: Rendering the data in the UI
+
+With the data decoded, it’s time to display it in the UI. Update your React
+component to process the result and render the feed:
+
+```reason
+  {switch (data) {
+    | Loading => <div> {React.string("Loading...")} </div>
+    | Loaded(Error(msg)) => <div> {React.string(msg)} </div>
+    | Loaded(Ok(feed)) =>
+      <div>
+        <h1> {React.string("GitHub Feed")} </h1>
+        <ul>
+          {feed.entries
+          |> Array.map(entry =>
+                <li key={string_of_float(entry.Feed.updated)}>
+                  <h2> {React.string(entry.title)} </h2>
+                  {switch (entry.content) {
+                  | None => React.null
+                  | Some(content) =>
+                    <p dangerouslySetInnerHTML={"__html": content} />
+                  }}
+                  <ul>
+                    {entry.links
+                    |> Array.map((link: Feed.link) =>
+                          <li>
+                            <a href={link.href}>
+                              {React.string(link.title)}
+                            </a>
+                          </li>
+                        )
+                    |> React.array}
+                  </ul>
+                </li>
+              )
+          |> React.array}
+        </ul>
+      </div>
+    }}
+```
+
+Note how we leverage pattern matching (`switch`) together with variants to map
+the data state directly to the UI.
+
+Also, unlike in JavaScript, plain strings cannot be used directly as React
+elements in ReasonReact; they need to be converted using `React.string`.
+
+Lastly, we use specific type annotations in `Array.map` (e.g., `Feed.link`) to
+assist the compiler in inferring the correct type within the callback. This
+isn’t always necessary, but it’s often required in mapping functions due to the
+way type inference and piping work in OCaml.
+
+### 7.3: Step 7 completion check
+
+Reload your application, and you should now see the GitHub feed displayed on the
+page. The data is fetched, decoded, and rendered dynamically. If everything
+works, the console should be free of errors, and the feed should be visible.
+
+Can you try updating the rendering function to also show `entry.content` and
```

Comment:
Didn't you already render those in the code above?

Reply:
right... removed in https://github.com/ahrefs/react-alicante-workshop/pull/5/commits/c8821ee89764f0115935f7de951f476dee44578d.